File: src/main/java/de/cf/autoscaler/prometheus/kafka/ScalingLogConsumer.java
package de.cf.autoscaler.prometheus.kafka;


import com.google.protobuf.InvalidProtocolBufferException;

import de.cf.autoscaler.kafka.AutoScalerConsumer;
import de.cf.autoscaler.kafka.ByteConsumerThread;
import de.cf.autoscaler.kafka.messages.ScalingLog;
import de.cf.autoscaler.kafka.protobuf.ProtobufScalingWrapper.ProtoScaling;

public class ScalingLogConsumer implements AutoScalerConsumer {
	
	private ByteConsumerThread consThread;
	private InfluxDBWriter writer;
	
	public ScalingLogConsumer(String topic, String groupId, String hostname, int port, InfluxDBWriter writer) {
		consThread = new ByteConsumerThread(topic, groupId, hostname, port, this);
		this.writer = writer;
		
	}
	
	public void startConsumer() {
		consThread.start();
	}
	
	public void stopConsumer() {
		consThread.getKafkaConsumer().wakeup();
	}
	
	public void consume(byte[] bytes) {
		try {
			ScalingLog log = new ScalingLog(ProtoScaling.parseFrom(bytes));
			writer.writeScalingLog(log);

		} catch (InvalidProtocolBufferException e) {
			e.printStackTrace();
		}
	}
	
	public String getType() {
		return "quotient";
	}

}

File: prometheus-writer/src/main/java/de/evoila/cf/prometheus/writer/kafka/ScalingLogConsumer.kt
package de.evoila.cf.prometheus.writer.kafka

import com.google.protobuf.InvalidProtocolBufferException

import de.evoila.cf.autoscaler.kafka.AutoScalerConsumer
import de.evoila.cf.autoscaler.kafka.ByteConsumerThread
import de.evoila.cf.autoscaler.kafka.KafkaPropertiesBean
import de.evoila.cf.autoscaler.kafka.messages.ScalingLog
import de.evoila.cf.autoscaler.kafka.protobuf.PbScalingLog.ProtoScalingLog
import de.evoila.cf.prometheus.PrometheusWriter

class ScalingLogConsumer(groupId: String, kafkaPropertiesBean: KafkaPropertiesBean,
                         private val writer: PrometheusWriter) : AutoScalerConsumer {

    private val consThread: ByteConsumerThread = ByteConsumerThread(kafkaPropertiesBean.metricContainerTopic,
            groupId,
            kafkaPropertiesBean.host,
            kafkaPropertiesBean.port, this)

    override fun startConsumer() {
        consThread.start()
    }

    override fun stopConsumer() {
        consThread.kafkaConsumer.wakeup()
    }

    override fun consume(bytes: ByteArray) {
        try {
            val log = ScalingLog(ProtoScalingLog.parseFrom(bytes))
            writer.writeScalingLog(log)

        } catch (e: InvalidProtocolBufferException) {
            e.printStackTrace()
        }

    }

    override fun getType(): String {
        return "quotient"
    }

}
